src/coreclr/vm/olevariant.cpp

```diff
@@ -1176,52 +1169,16 @@ void SafeVariantClear(VARIANT* pVar)
     if (pVar)
     {
         GCX_PREEMP();
-        SCAN_EHMARKER();
-        PAL_CPP_TRY
-        {
-            // These are holders to tell the contract system that we're catching all exceptions.
-            SCAN_EHMARKER_TRY();
-            CLR_TRY_MARKER();
-
-            // Most of time, oleaut32.dll is loaded already when we get here.
-            // Sometimes, CLR initializes Variant without loading oleaut32.dll, e.g. VT_BOOL.
-            // It is better for performance with EX_TRY than
-
-            SafeVariantClearHelper(pVar);
-
-            SCAN_EHMARKER_END_TRY();
-        }
-#pragma warning(suppress: 4101)
-        PAL_CPP_CATCH_DERIVED(OutOfMemoryException, obj)
-        {
-            SCAN_EHMARKER_CATCH();
-
-#if defined(STACK_GUARDS_DEBUG)
-            // Catching and just swallowing an exception means we need to tell
-            // the SO code that it should go back to normal operation, as it
-            // currently thinks that the exception is still on the fly.
-            GetThread()->GetCurrentStackGuard()->RestoreCurrentGuard();
-#endif
-
-            SCAN_EHMARKER_END_CATCH();
-        }
-        PAL_CPP_ENDTRY;
-
+        VariantClear(pVar);
         FillMemory(pVar, sizeof(VARIANT), 0x00);
```

Comment:
```suggestion
        FillMemory(pVar, sizeof(VARIANT), 0x00);
        V_VT(pVar) = VT_EMPTY;
```

This is sort of an odd thing. The `vt` field is being overwritten here. When `VariantClear` is called the instance is reset to `VT_EMPTY`. We should either explicit reset it to `VT_EMPTY` or comment that since `VT_EMPTY` is 0 this is ensuring that value remains.

Reply:
Yes, it is odd. It would be best to delete it, but it is legacy COM interop where users depend on every undocumented detail.